packages/components/pie-input/src/index.ts

```diff
@@ -27,24 +27,44 @@ export class PieInput extends FormControlMixin(RtlMixin(LitElement)) implements
 
     @property({ type: String, reflect: true })
     @validPropertyValues(componentSelector, types, 'text')
-    public type: InputProps['type'] = 'text';
+    public type? = InputDefaultPropertyValues.type;
 
     @property({ type: String })
-    public value = '';
+    public value? = InputDefaultPropertyValues.value;
 
     @property({ type: String })
-    public name = '';
+    public name?: InputProps['name'];
+
+    @property({ type: String })
+    public pattern?: InputProps['pattern'];
+
+    @property({ type: Number })
+    public minlength?: InputProps['minlength'];
+
+    @property({ type: Number })
+    public maxlength?: InputProps['maxlength'];
+
+    @query('input')
+    private input?: HTMLInputElement;
+
+    /**
+     * (Read-only) returns a ValidityState with the validity states that this element is in.
+     * https://developer.mozilla.org/en-US/docs/Web/API/HTMLObjectElement/validity
+     */
+    public get validity (): ValidityState {
+        return (this.input as HTMLInputElement).validity;
+    }
 
     protected firstUpdated (_changedProperties: PropertyValues<this>): void {
         super.firstUpdated(_changedProperties);
-        this._internals.setFormValue(this.value);
+        this._internals.setFormValue(this.value as string);
```

Comment:
Does this handle `undefined` nicely?

Reply:
Not currently simply because it either defaults to an empty string or it is driven off of the value typed in. I can write some tests in a follow up PR and add some more defensive coding to handle this situation. I am unsure how you'd ever get undefined unless we can pass it from a SPA

Reply:
Tbh I'm not quite sure what we'd want to occur in that situation. I think passing undefined up and letting it error is likely the preferred behaviour rather than the input trying to self-correct